tests/export/image_resize/test_image_resize.py

```diff
@@ -0,0 +1,175 @@
+
+
+class ImageResizerTestCase(TestCase):
+    def test_init(self):
+        image_data = get_img('image1.png', as_binary=True)
+
+        ir = ImageResizer(image_data, 'image1.png', '100 px', '100 px')
+
+        self.assertTrue(ir)
+
+    def test_init_object(self):
+        image_data = get_img('image1.png', as_binary=True)
+        ir = ImageResizer(image_data, 'image1.png', '100 px', '100 px')
+
+        self.assertTrue(ir)
+
+    def test_has_skipable_extension_tue(self):
+        image_data = get_img('image1.png', as_binary=True)
+        ir = ImageResizer(image_data, 'image1.png', '100 px', '100 px', skip_extensions=['png'])
+
+        self.assertTrue(ir.has_skipable_extension())
+
+    def test_has_skipable_extension_false(self):
+        image_data = get_img('image1.png', as_binary=True)
+        ir = ImageResizer(image_data, 'image1.png', '100 px', '100 px')
+
+        self.assertFalse(ir.has_skipable_extension())
+
+    def test_has_height_and_width_true(self):
+        image_data = get_img('image1.png', as_binary=True)
+        ir = ImageResizer(image_data, 'image1.png', '10 px', '20 px')
+
+        self.assertTrue(ir.has_height_and_width())
+
+    def test_has_height_and_width_false(self):
+        image_data = get_img('image1.png', as_binary=True)
+
+        ir = ImageResizer(image_data, 'image1.png', '0 px', '10 px')
+        self.assertFalse(ir.has_height_and_width())
+
+        ir = ImageResizer(image_data, 'image1.png', '10 px', '0 px')
+        self.assertFalse(ir.has_height_and_width())
+
+        ir = ImageResizer(image_data, 'image1.png', '0 px', '0 px')
+        self.assertFalse(ir.has_height_and_width())
+
+    def test_init_image(self):
+        image_data = get_img('image1.png', as_binary=True)
+        ir = ImageResizer(image_data, 'image1.png', '100 px', '100 px')
+        ir.init_image()
+        self.assertEqual(ir.image, Image.open(StringIO(image_data)))
+
+        image_data = get_img('image1.data', as_binary=True)
+        ir = ImageResizer(image_data, 'image1.png', '100 px', '100 px')
+        ir.init_image()
+        # here the final Image object should be the same as we create it from binary data
+        self.assertEqual(ir.image, Image.open(StringIO(get_img('image1.png', as_binary=True))))
+
+    def test_init_image_exception(self):
+        ir = ImageResizer(b'test_data3434', 'image1.png', '100 px', '100 px')
+
+        self.assertRaises(IOError, ir.init_image)
+
+    def test_resize_image_skip(self):
+        image_data = get_img('image1.data', as_binary=True)
+
+        ir = ImageResizer(image_data, 'image1.png', '48 px', '48 px')
+        ir.init_image()
+
+        result = ir.resize_image()
+
+        self.assertIsNone(result)
+
+    def test_resize_image_success(self):
+        image_data = get_img('image1.png', as_binary=True)
+
+        ir = ImageResizer(image_data, 'image1.png', '48 px', '48 px')
+
+        ir.init_image()
+        result = ir.resize_image()
```

Comment:
Well, first of all yes it should be consistent and return only True|False. The diff is that in the first case we test as input image1.data and in such cases we skip resizing the image.